src/Ball.js

```diff
@@ -127,7 +129,8 @@ export class Ball {
   }
 
   isAwareToBeInfected (otherBall) {
-    return this.hasAppInstalled && otherBall.hasAppInstalled
+    return this.hasAppInstalled && otherBall.hasAppInstalled &&
+      this.sketch.random(100) >= this.appFailurePercentage
```

Comment:
consider to store this boolean value as Ball member variable in place of `appFailurePercentage`.
If in the future we will reuse the function in a different part of the code we may get wrong results (different awareness).

Reply:
I don't think it would be the same, because each collision should be affected by a different shot: sometimes the app could work, sometimes doesn't. Am I right?

Reply:
I understand your point and generally speaking you're right, the app failure must be re-evaluated at each collision. 

BTW, this opens a new point:
the code invokes the isAwareToBeInfected just once per ball (look few lines above, if the ball's state is `well` only).
Is this the expected behavior? The ball should re-check for awareness every time it collides with another infected ball. Otherwise, if the first time the app fails, the ball will continue to move even after any next collision with an infected ball (the app may work the next times).

But we can open a new issue, so I would accept the current behavior w/o any objection.